src/GafferScene/SceneAlgo.cpp

```diff
@@ -493,7 +498,10 @@ class CapturingMonitor : public Monitor
 		{
 			if(
 				processType == g_hashProcessType &&
-				plug->parent<ScenePlug>() && plug->getName() == m_scenePlugChildName
+				(
+					( plug->parent<ScenePlug>() && plug->getName() == m_scenePlugChildName ) ||
+					( (Gaffer::TypeId)plug->typeId() == Gaffer::TypeId::ObjectPlugTypeId && plug->getName() == g_processedObjectPlugName )
```

Comment:
Could we put this conditional in a little inline `shouldCapture( const Plug * )` function, and then use `!shouldCapture()` in `processStarted()`? The logic seems subtle enough and likely enough to get out of sync that it might be worth not transposing the individual elements to the negative in `processStarted()`.

I wonder if we should also check that `plug->parent<ObjectProcessor>()` in the `__processedObject` case?

Reply:
Definitely an improvement, fixed and squashed in to https://github.com/GafferHQ/gaffer/pull/5519/commits/823b997ecac10932bf89f4c15e9413b375c8a9a0